presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/DockerContainer.java

```diff
@@ -298,6 +299,13 @@ public DockerContainer waitingForAll(WaitStrategy... strategies)
         return this;
     }
 
+    @Override
+    public DockerContainer waitingFor(WaitStrategy waitStrategy)
+    {
+        checkState(this.waitStrategy == null, "waitStrategy is already defined");
```

Comment:
i am concerned this will prevent us from reconfiguring a wait strategy along when reconfiguring the container. For example, there is Presto container with some Presto wait strategy using CLI and we now modify the container securing the frontend and we need to amend the wait strategy.

Also, we allow overriding of pretty anything (files, settings, etc), why not wait strategy?

What is the concern you're guarding against here?

Reply:
I've seen (and corrected) invocations where there was multiple, consecutive waitingFor calls replacing each other (instead waitingForAll should be used). Other solution to this is to merge wait strategies on consecutive calls.

Reply:
Thanks for explanation. so we can add `waitingFor` override as you did (with a check) and `overrideWaitingFor` to skip the check?
we can (should?) delay adding the new method until needed though

Reply:
I approached that differently. PTAL now